File: frontend/src/store/api/result/dto/product/variant/AmplifiersVariantOutputDTO.ts
import { ProductVariantOutputDTO } from "./ProductVariantOutputDTO";

export interface AmplifiersVariantOutputDTO extends ProductVariantOutputDTO {
    color: string,
    amplifiersId: number
};

export function isAmplifiersVariant(variant: ProductVariantOutputDTO): variant is AmplifiersVariantOutputDTO {
    return "amplifiersId" in variant;
}

export function isAmplifiersVariantArray(variants: ProductVariantOutputDTO[]): variants is AmplifiersVariantOutputDTO[] {
    return variants.every(isAmplifiersVariant);
}

File: backend/product-client/src/main/java/org/andante/amplifiers/dto/AmplifiersVariantOutputDTO.java
package org.andante.amplifiers.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import org.andante.product.dto.ProductVariantOutputDTO;

@SuperBuilder
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class AmplifiersVariantOutputDTO extends ProductVariantOutputDTO {

    private String color;
    private Long amplifiersId;
}
